test/end2end_test.go

```diff
@@ -7819,3 +7819,36 @@ func (s) TestStreamingServerInterceptorGetsConnection(t *testing.T) {
 		t.Fatalf("ss.Client.StreamingInputCall(_) = _, %v, want _, %v", err, io.EOF)
 	}
 }
+
+func unaryInterceptorVerifyPost(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return nil, status.Error(codes.NotFound, "metadata was not in context")
+	}
+	method := md.Get(":method")
+	if len(method) != 1 {
+		return nil, status.Error(codes.NotFound, ":method value had more than one value")
```

Comment:
This and the next shouldn't use NotFound.
Use InvalidArgument?

Reply:
Switched.